File: leetcode/easy/python/lc_617.py
def mergeTrees(root1: Optional[TreeNode], root2: Optional[TreeNode]) -> Optional[TreeNode]:
        return self.dfs(root1, root2)
        
    def traverse(self, node1, node2):
        if node1 and node2:
            root = TreeNode(node1.val + node2.val)
            root.left = self.traverse(node1.left, node2.left)
            root.right = self.traverse(node1.right, node2.right)
            return root
        return node1 or node2

File: leetcode/easy/java/lc617.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode mergeTrees(TreeNode root1, TreeNode root2) {
        if (root1 == null) {
            return root2;
        } 
        else if (root2 == null) {
            return root1;
        } 
        else {
            TreeNode merged_tree = new TreeNode(root1.val + root2.val);
            merged_tree.left = mergeTrees(root1.left, root2.left);
            merged_tree.right = mergeTrees(root1.right, root2.right);
            return merged_tree;
        }
    }
}

